File: backend/src/main/java/be/technobel/parsemaster/security/JwtAuthenticationFilter.java
package be.technobel.parsemaster.security;

import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;
import java.io.IOException;

@Component
public class JwtAuthenticationFilter extends OncePerRequestFilter {
  private final JWTProvider jwtProvider;

  public JwtAuthenticationFilter(JWTProvider jwtProvider) {
    this.jwtProvider = jwtProvider;
  }

  @Override
  protected void doFilterInternal(
    HttpServletRequest request,
    HttpServletResponse response,
    FilterChain filterChain
  ) throws ServletException, IOException {
    final var token = jwtProvider.extractToken(request);
    if(token != null) {
      if(jwtProvider.validateToken(token) && jwtProvider.isNotRevoked(token)) {
        final var auth = jwtProvider.createAuthentication(token);
        SecurityContextHolder.getContext().setAuthentication(auth);
      }
    }
    filterChain.doFilter(request, response);
  }
}

File: src/main/kotlin/be/technobel/playzone/pl/security/JwtAuthenticationFilter.kt
package be.technobel.playzone.pl.security

import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter

@Component
class JwtAuthenticationFilter(private val jwtProvider: JWTProvider) : OncePerRequestFilter() {
	override fun doFilterInternal(
		request: HttpServletRequest,
		response: HttpServletResponse,
		filterChain: FilterChain
	) {
		val token = jwtProvider.extractToken(request)
		if(token != null && jwtProvider.validateToken(token) && !jwtProvider.isRevoked(token)) {
			val auth = jwtProvider.createAuthentication(token)
			SecurityContextHolder.getContext().authentication = auth
		}
		filterChain.doFilter(request, response)
	}
}